[Bug] Expansion of RGB_DISABLE_TIMEOUT overflows

<!-- Provide a general summary of the bug in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Describe the Bug

<!-- A clear and concise description of what the bug is. -->
User is able to define `RGB_DISABLE_AFTER_TIMEOUT` in their config which roughly equates to a number of seconds of inactivity before RGB lights are disabled. This bug limits this to a maximum 27, as anything higher causes an overflow and compiler error.

```
Compiling: quantum/rgb_matrix.c
quantum/rgb_matrix.c: In function 'rgb_matrix_task':
quantum/rgb_matrix.c:61:60: error: integer overflow in expression of type 'int' results in '-31936' [-Werror=overflow]
 #    define RGB_DISABLE_TIMEOUT (RGB_DISABLE_AFTER_TIMEOUT * 1200)
                                                            ^
quantum/rgb_matrix.c:392:39: note: in expansion of macro 'RGB_DISABLE_TIMEOUT'
         (rgb_anykey_timer > (uint32_t)RGB_DISABLE_TIMEOUT) ||
                                       ^~~~~~~~~~~~~~~~~~~

```

The code above expands to`(uint32_t)(28 * 1200)` (if `RGB_DISABLE_AFTER_TIMEOUT` is defined as `28`) with the cast outside the parenthesis. ~Removing the parenthesis on line 61 fixes this issue and compiles without problem, but I am not sure if it is the most correct solution as this macro is used elsewhere.~  
~~#    define RGB_DISABLE_TIMEOUT RGB_DISABLE_AFTER_TIMEOUT * 1200~~ I'm dumb.

## System Information

 - Keyboard:  dz65rgb
   - Revision:  v2
 - Operating system: Win10
 - AVR GCC version: 8.3.0
 - ARM GCC version: 8.3.1 20190703
 - QMK Firmware version: 0.9.46

--- a/quantum/rgb_matrix.c
+++ b/quantum/rgb_matrix.c
@@ -58,7 +58,7 @@ const point_t k_rgb_matrix_center = RGB_MATRIX_CENTER;
 // ------------------------------------------
 
 #if defined(RGB_DISABLE_AFTER_TIMEOUT) && !defined(RGB_DISABLE_TIMEOUT)
-#    define RGB_DISABLE_TIMEOUT (RGB_DISABLE_AFTER_TIMEOUT * 1200ul)
+#    define RGB_DISABLE_TIMEOUT (RGB_DISABLE_AFTER_TIMEOUT * 1200UL)
 #endif
 
 #ifndef RGB_DISABLE_TIMEOUT

--- a/quantum/rgb_matrix.c
+++ b/quantum/rgb_matrix.c
@@ -58,7 +58,7 @@ const point_t k_rgb_matrix_center = RGB_MATRIX_CENTER;
 // ------------------------------------------
 
 #if defined(RGB_DISABLE_AFTER_TIMEOUT) && !defined(RGB_DISABLE_TIMEOUT)
-#    define RGB_DISABLE_TIMEOUT (RGB_DISABLE_AFTER_TIMEOUT * 1200)
+#    define RGB_DISABLE_TIMEOUT (RGB_DISABLE_AFTER_TIMEOUT * 1200ul)
 #endif
 
 #ifndef RGB_DISABLE_TIMEOUT

